### ISSUE TITLE
Combining Multiple Annotations using Streams (Piped input) fails

### ISSUE THREAD

**Issue by annblackz** [samtools/bcftools#379]
Hi, I am trying to chain together multiple annotations using streams, for example:

bcftools annotate -Ou -a whole_genome_SNVs.add-chr.otoscopev6v7.tsv.gz -c CHROM,POS,ID,REF,ALT,CADD_RAW,CADD_PHRED -h CADD.vcf.header GJB2.CEPH.104_1-12_2-1330_1.unifiedGenotyper.vcf.gz | bcftools annotate -a ExAC.r0.3.sites.vep.vcf.gz 

When doing this, I always receive an error:
Failed to open -: not compressed with bgzip

Is this type of operation not supported, or am I making a mistake in how I am running the annotations?  Running each annotation independent of one another seems to work. 

Thanks so much for your help!

**Comment by pd3** [comment]
Hi, sorry about the cryptic error message which comes from htslib internals. When the `-a` file is a VCF, the annotation does not work on streams. This is because a different mechanism is used to keep the two files in sync. This can be improved in htslib, I'll mark this issue as a feature request.

**Comment by annblackz** [comment]
Thanks for your response :+1:  I agree it would be desirable

**Comment by tommycarstensen** [comment]
I agree this would be desirable 👍 Otherwise I can't carry over FORMAT fields from two VCFs to a third VCF without looping over the records twice. Thanks!

**Comment by NeillGibson** [comment]
Hi @pd3  can I ask what the status is of this issue.  I run into the same issue with bcftools 1.3. 

We are trying  to annotate user defined exports on the fly using different annotation VCF file.

**Comment by pd3** [comment]
Hi, it would be a nice feature to have, but it is not coming any time soon, there are more pressing issues. If it was added, it never would be by storing the whole VCF in memory, that does not scale.

**Comment by NeillGibson** [comment]
Ok thank you good too know. We will use the tab file based annotation option as an alternative solution.

**Comment by freeseek** [comment]
Could a better error message be provided? It seems like many users had the exact same issue. Also, I am trying to understand how to format the `-a` file as a TSV file, but I keep getting the error `The tag "XXX" is not defined in xxx.tsv`

**Comment by dmckean** [comment]
If the annotations are site-level, it may be a suitable workaround to combine these annotations on a VCF where the genotypes have been dropped. It isn't much more efficient, but you would save on disk by waiting to annotate the genotyped VCF from a single, annotated VCF.
